dask/tests/test_threaded.py

```diff
@@ -105,58 +103,28 @@ def test_f():
     assert L == [1] * 20
 
 
-code = """
-import sys
-import time
-from dask.threaded import get
-
-def signal_started():
-    sys.stdout.write('started\\n')
-    sys.stdout.flush()
-
-def long_task(x):
-    time.sleep(5)
-
-dsk = {('x', i): (long_task, 'started') for i in range(100)}
-dsk['started'] = (signal_started,)
-dsk['x'] = (sum, list(dsk.keys()))
-get(dsk, 'x')
-"""
-
-
 # TODO: this test passes locally on windows, but fails on appveyor
 # because the ctrl-c event also tears down their infrastructure.
 # There's probably a better way to test this, but for now we'll mark
 # it slow (slow tests are skipped on appveyor).
 @pytest.mark.slow
 def test_interrupt():
+
+    def long_task():
+        sleep(5)
+
+    def interrupt():
+        sleep(0.5)
+        pid = os.getpid()
+        os.kill(pid, signal.SIGINT)
```

Comment:
I see. Knowing that. Why not change this to [`thread.interrupt_main`]( https://docs.python.org/2/library/thread.html#thread.interrupt_main ), which has exactly this behavior and is a bit cleaner? This may even clear up the intermittent failures of this new approach.